### ISSUE TITLE
The `@stylistic/unit-case` rule cuts a word at an escaped star, and names the backslash left behind as part of a unit

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#414]
Found while writing the branch that closes [#297](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/297). Reproduces on `main` before that branch as it does after it.

### What minimal example or steps are needed to reproduce the bug?

A word in which the star is escaped, so that it is a character of an identifier rather than an operator:

```css
a { b: 10PX\*2REM; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/unit-case": "lower"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json
```

### What did you expect to happen?

Nothing said about a word no browser reads as a dimension, or — since this rule recases a unit no specification knows — one warning naming the one unit the word holds, `PX\*2REM`.

### What actually happened?

```
1:10-1:14  Expected "PX\" to be "px\"
1:16-1:19  Expected "REM" to be "rem"
```

`PX\` is no unit of anything: the word was cut at a star that is no operator, and the backslash that escapes it was left on the end of the first piece. `REM` is named as a unit of its own, and there is no second dimension for it to be the unit of.

### What is the cause?

`lib/rules/unit-case/index.js` splits such a word at every star it holds:

```js
for (let part of value.split(`*`)) {
```

`String.prototype.split` knows nothing of escaping, so `10PX\*2REM` becomes `10PX\` and `2REM`, and each piece is read as a dimension of its own.

`lightningcss@1.33.0` reads the two words differently, which is what says the split is wrong rather than merely untidy — it recases the units of the unescaped word and leaves the escaped one exactly as it was:

| input | what `lightningcss` gives back |
| --- | --- |
| `a { b: 10PX*2REM; }` | `a { b: 10px*2rem; }` |
| `a { b: 10PX\*2REM; }` | `a { b: 10PX\*2REM; }` |

### Do you have a proposal to fix the bug?

Split at a star the file spells as an operator and not at one it escapes — a star with an odd number of backslashes in front of it is a character of the identifier it stands in, and no boundary between two dimensions.

**Comment by firefoxic** [comment]
One more shape this reading owns, noted from the branch for #426. To `@csstools/css-tokenizer` an escaped hash is a code point of the identifier, so `10px\#fff` is one dimension token with the unit `px\#fff`, and `lightningcss` 1.33.0 prints it as it stands. `getDimension` used to take the hash out and name `px\fff`; since that branch the copy it reads ends at the first character an interpolation is spelled with, escaped or not, so the unit named is `px\` — the same trailing backslash as `PX\` above, and for the same reason: the reading knows nothing of an escape. `scripts/sweeps/hash-after-dimension.ts` carries the spelling under the name `escapedHash`, behind fourteen heads, in front of them and inside a multiplication.
